### DISCUSSION TITLE
Add a progress bar

### DISCUSSION THREAD

**Discussion by khamaileon** [valhalla/valhalla#4160] [Ideas]
There are a lot of time-consuming processes in Valhalla, and I think it would be interesting to have a progress bar to see how things are going. What do you think? I've been thinking about this library: https://github.com/p-ranav/indicators.

**Comment by nilsnolde** [comment]
That'd be nice tbh.. it's a bit hard often (e.g. tile building) to come up with a proper heuristic of "progress" (I guess that's almost always the case in any non-trivial process..), but IMO it'd be doable. However, I do think that the (often too) frequent logging gives an indication that _something_ is happening. Doesn't tell you how far along you are in the process of course.. Personally I'm open to the idea. Would you generally have interest to draft a PR with one executable to start with (valhalla_ingest_traffic might actually be a good one, it doesn't branch out too much)?

**Reply by khamaileon** [reply]
Yes, I was actually thinking about valhalla_ingest_traffic. It would be a first step towards improving execution speed.

**Comment by nilsnolde** [comment]
FWIW, OSRM does have a small home-baked progress bar for some internal processes and I do enjoy that when seldom running it. Yeah I think I really do like the idea. It'd need to be optional though and if wanted, we should probably suppress our own logging or at least defer it to stderr.

**Reply by khamaileon** [reply]
Thanks, I'll take a look at OSRM.
